dist/images/ovndb-raft-functions.sh

```diff
@@ -238,27 +273,63 @@ ovsdb-raft() {
     ovn_raft_conn_ip_url_suffix=":[::]"
   fi
 
-  if [[ "${POD_NAME}" == "ovnkube-db-0" ]]; then
-    run_as_ovs_user_if_needed \
-      ${OVNCTL_PATH} run_${db}_ovsdb --no-monitor \
-      --db-${db}-cluster-local-addr=[${ovn_db_host}] \
-      --db-${db}-cluster-local-port=${raft_port} \
-      --db-${db}-cluster-local-proto=${transport} \
-      ${db_ssl_opts} \
-      --ovn-${db}-log="${ovn_loglevel_db}" &
-  else
-    # join the remote cluster node if the DB is not created
-    if [[ "${initialize}" == "true" ]]; then
-      wait_for_event ready_to_join_cluster ${db} ${port}
-    fi
-    run_as_ovs_user_if_needed \
+  initial_raft_create=true
+  if [[ "${initialize}" == "true" ]]; then
+    # check to see if a cluster already exists. If it does, just join it.
+    counter=0
+    cluster_found=false
+    while [ $counter -lt 5 ]; do
+      if cluster_exists ${db} ${port}; then
+        cluster_found=true
+        break
+      fi
+      sleep 1
+      counter=$((counter+1))
+    done
+
+    if ${cluster_found}; then
+      echo "Cluster already exists for DB: ${db}"
+      initial_raft_create=false
+      run_as_ovs_user_if_needed \
       ${OVNCTL_PATH} run_${db}_ovsdb --no-monitor \
-      --db-${db}-cluster-local-addr=[${ovn_db_host}] --db-${db}-cluster-remote-addr=[${init_ip}] \
+      --db-${db}-cluster-local-addr=$(bracketify ${ovn_db_host}) --db-${db}-cluster-remote-addr=${init_ip} \
       --db-${db}-cluster-local-port=${raft_port} --db-${db}-cluster-remote-port=${raft_port} \
       --db-${db}-cluster-local-proto=${transport} --db-${db}-cluster-remote-proto=${transport} \
       ${db_ssl_opts} \
       --ovn-${db}-log="${ovn_loglevel_db}" &
+    else
+      # either we need to initialize a new cluster or wait for db-0 to create it
+      if [[ "${POD_NAME}" == "ovnkube-db-0" ]]; then
+        echo "Cluster does not exist for DB: ${db}, creating new raft cluster"
```

Comment:
If we do have all the raft dbs down, we will try to create a new cluster and all the new dbs will have different db-ids. The ovn-k master (leader) will detect that at some point that all the raft db-ids have departed from what's in the cache and restart. This will force the sync of nbdb and sbdb raft nodes. I think this is a corner case that can happen, but I feel like it's covered.